hledger-lib/Hledger/Reports/ReportOptions.hs

```diff
@@ -488,13 +491,56 @@ flat_ = not . tree_
 -- depthFromOpts :: ReportOpts -> Int
 -- depthFromOpts opts = min (fromMaybe 99999 $ depth_ opts) (queryDepth $ queryFromOpts nulldate opts)
 
--- | Convert this journal's postings' amounts to cost using their
--- transaction prices, if specified by options (-B/--cost).
--- Maybe soon superseded by newer valuation code.
+-- | Convert this journal's postings' amounts to cost using their transaction prices,
+-- if specified by options (-B/--cost). Strip prices if not needed.
 journalSelectingAmountFromOpts :: ReportOpts -> Journal -> Journal
-journalSelectingAmountFromOpts opts = case cost_ opts of
+journalSelectingAmountFromOpts ropts = maybeStripPrices . case cost_ ropts of
     Cost   -> journalToCost
     NoCost -> id
+  where
+    maybeStripPrices = if show_costs_ ropts then id else journalMapPostingAmounts mixedAmountStripPrices
+
+-- | Convert this journal's postings' amounts to cost using their transaction
+-- prices and apply valuation, if specified by options (-B/--cost). Strip prices
```

Comment:
Would something like this be clearer: "Convert this journal's postings' amounts to cost and/or to value, if specified by options (-B/--cost/-V/-X/--value etc.) ..." 
I think I saw similar wording in some other new/updated functions as well, it had me thinking they just converted to cost.

Reply:
Otherwise, LGTM.

Reply:
I've updated the wording of that comment.